src/services/db_interface.ts

```diff
@@ -232,18 +255,88 @@ export interface ArtifactModel {
   aid: string;
   /** Required. copr-build */
   type: string;
-  payload:
-    | PayloadBrewBuild
-    | PayloadKojiBuild
-    | PayloadDistGitPR
-    | PayloadRedHatModule
-    | PayloadFedoraModule
-    | PayloadProductMDCompose;
+  payload?: {};
   states: ArtifactState[];
   /** When the mongodb-document will be auto-removed, for example: scratch build */
   expire_at?: Date;
 }
 
+export type ArtifactTypeBrewBuild = 'brew-build';
+export type ArtifactTypeKojiBuild = 'koji-build';
+export type ArtifactTypeDistGitPR = 'dist-git-pr';
+export type ArtifactTypeRedHatModule = 'redhat-module';
+export type ArtifactTypeFedoraModule = 'fedora-module';
+export type ArtifactTypeProductMDCompose = 'productmd-compose';
+export type ArtifactTypeRedhatContainerImage = 'redhat-container-image';
+
+export type ArtifactModelBrewBuild = ArtifactBaseModel & {
+  type: ArtifactTypeBrewBuild;
+  payload: PayloadBrewBuild;
+};
+
+export type ArtifactModelKojiBuild = ArtifactBaseModel & {
+  type: ArtifactTypeKojiBuild;
+  payload: PayloadKojiBuild;
+};
+
+export type ArtifactModelDistGitPR = ArtifactBaseModel & {
+  type: ArtifactTypeDistGitPR;
+  payload: PayloadDistGitPR;
+};
+
+export type ArtifactModelRedHatModule = ArtifactBaseModel & {
+  type: ArtifactTypeRedHatModule;
+  payload: PayloadRedHatModule;
+};
+
+export type ArtifactModelFedoraModule = ArtifactBaseModel & {
+  type: ArtifactTypeFedoraModule;
+  payload: PayloadFedoraModule;
+};
+
+export type ArtifactModelProductMDCompose = ArtifactBaseModel & {
+  type: ArtifactTypeProductMDCompose;
+  payload: PayloadProductMDCompose;
+};
+
+export type ArtifactModelRedhatContainerImage = ArtifactBaseModel & {
+  type: ArtifactTypeRedhatContainerImage;
+  payload: PayloadContainerImage;
+};
+
+export type ArtifactModel =
+  | ArtifactModelBrewBuild
+  | ArtifactModelKojiBuild
+  | ArtifactModelDistGitPR
+  | ArtifactModelRedHatModule
+  | ArtifactModelFedoraModule
+  | ArtifactModelProductMDCompose
+  | ArtifactModelRedhatContainerImage;
+
+/**
+ * TypeScript guards
+ */
+export function isArtifactBrewBuild(
+  artifact: ArtifactModel,
+): artifact is ArtifactModelBrewBuild {
+  return artifact.type === 'brew-build';
+}
+export function isArtifactRedHatModule(
+  artifact: ArtifactModel,
+): artifact is ArtifactModelRedHatModule {
+  return artifact.type === 'redhat-module';
+}
+export function isArtifactCompose(
+  artifact: ArtifactModel,
+): artifact is ArtifactModelProductMDCompose {
+  return artifact.type === 'productmd-compose';
+}
+export function isArtifactRedhatContainerImage(
```

Comment:
This naming is inconsistent with the previous. We have `isArtifactRedHatModule`, but `isArtifactRedhatContainerImage` – note the `RedHat` vs. `Redhat`.